Scenarios/Logging/LoggingSample/IService.cs

```diff
@@ -0,0 +1,72 @@
+﻿using CoreWCF;
+using System;
+using System.Runtime.Serialization;
+
+namespace LoggingSample
+{
+    [ServiceContract]
+    public interface IService
+    {
+        [OperationContract]
+        string GetData(int value);
+
+        [OperationContract]
+        CompositeType GetDataUsingDataContract(CompositeType composite);
+    }
+
+    public class Service : IService
+    {
+        private ILogger<Service> _logger;
+
+        // Parameterized constructor will be called by Dependency Injection
+        // Logs will be created under the name "LoggingSample.Service" as that's the type name used constructing the logger object
+        public Service(ILogger<Service> logger)
+        {
+            _logger = logger;
+        }
+
+        public string GetData(int value)
+        {
+            _logger.LogInformation("GetData called with value {value}", value);
+            return string.Format("You entered: {0}", value);
+        }
+
+        public CompositeType GetDataUsingDataContract(CompositeType composite)
+        {
```

Comment:
Is it worth demonstrating how to inject a different logger into an operation? Or is that overloading the sample with too much? To inject it into the operation, you make the `Service` class partial and change the implementation method like this:
```
public CompositeType GetDataUsingDataContract(CompositeType composite, [Injected]ILogger<CompositeType> localLogger)
```
CoreWCF back fills the method needed to match the interface, fetches the injected type from DI and then calls the implementation method.

Reply:
If we move the class to another file, then I think its separate enough to not confuse the other scenario. Presumably this pattern is about how WCF will construct data contract types, rather than the service contract.

Is logging the prototypical scenario for this, or are there other cases where you'd want a DI injected type?